core/domain/exp_domain.py

```diff
@@ -31,6 +31,7 @@
 from core.domain import html_validation_service
 from core.domain import interaction_registry
 from core.domain import param_domain
+import core.domain.state_domain as state_domain
```

Comment:
Ah, no, could you please do "from core.domain import state_domain", same as all the others?

Reply:
Done.

Reply:
Used import that way because of a cyclic dependency problem. Moving the constituents of state to state_domain solved it, thanks!